### PR TITLE
Create Remove All completed functionality

### PR DESCRIPTION
**Pull request by Reem-lab**
## changes that I updated on this milstone 
- Set Up with webpack template
 - Add functionality to `Clear All Completed` function.
 - change complete from false to true if Checked on checbox
 - using filter to remove 
 - Stroe data and remove it when remove from local storage  
 - no linters error

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,35 +1,163 @@
+/* eslint-disable eqeqeq */
```
Comment (arslanbisharat) [inline src/index.js]: - [x] Please don't disable linter rules. These are here to write clean and standard code.

File: src/index.js [hunk 2]
```diff
@@ -1,35 +1,163 @@
...
+// eslint-disable-next-line no-use-before-define
+getDataFromLocal();
 
-const task3 = {
-  description: 'Read articles',
-  completed: false,
-  index: 3,
+// submit tasks
+submit.onclick = () => {
+  // eslint-disable-next-line no-use-before-define
```
Comment (arslanbisharat) [inline src/index.js]: - [x] Please don't disable linter rules. These are here to write clean and standard code.

File: src/index.js [hunk 3]
```diff
@@ -1,35 +1,163 @@
...
+  input.value = ' '; // empty the input
 };
 
-const lists = [task1, task2, task3];
+// click on task elemnt to delete
+tasksDiv.addEventListener('click', (e) => {
+  // remove from page
+  if (e.target.classList.contains('del')) {
+    // remove from local
+    // eslint-disable-next-line no-use-before-define
```
Comment (arslanbisharat) [inline src/index.js]: - [x] Please don't disable linter rules. These are here to write clean and standard code.

File: src/index.js [hunk 4]
```diff
@@ -1,35 +1,163 @@
...
+    deleteTaskWith(e.target.parentElement.getAttribute('data-id'));
+    // remove from page
+    e.target.parentElement.remove();
+  }
+
+  // task element and update
+  if (e.target.classList.contains('checked')) {
+    // toggele completed for the task
+    // eslint-disable-next-line no-use-before-define
```
Comment (arslanbisharat) [inline src/index.js]: - [x] Please don't disable linter rules. These are here to write clean and standard code.

File: src/index.js [hunk 5]
```diff
@@ -1,35 +1,163 @@
...
+    id: Date.now(), // make it quall to time to be different
+    title: taskText,
+    completed: false,
+    index: arrayOfTasks.length,
+  };
+  // push my tasks to array
+  arrayOfTasks.push(task);
+
+  // Add elemnt to my page
+  // eslint-disable-next-line no-use-before-define
```
Comment (arslanbisharat) [inline src/index.js]: - [x] Please don't disable linter rules. These are here to write clean and standard code.

File: src/index.js [hunk 6]
```diff
@@ -1,35 +1,163 @@
...
+  addElementsToPageFrom(arrayOfTasks);
+
+  // add to local storage
+  // eslint-disable-next-line no-use-before-define
```
Comment (arslanbisharat) [inline src/index.js]: - [ ] Please don't disable linter rules. These are here to write clean and standard code.

File: src/index.js [hunk 7]
```diff
@@ -1,35 +1,163 @@
...
+    // create main div
+    const div = document.createElement('div');
+    div.className = 'task';
+
+    // check if task id done
+    if (task.completed) {
+      div.className = 'task done';
+    }
+    div.setAttribute('data-id', task.id);
+    // div.appendChild(document.createTextNode(task.title));
```
Comment (arslanbisharat) [inline src/index.js]: - [ ] Please removed commented code.

File: src/index.js [hunk 8]
```diff
@@ -1,35 +1,163 @@
...
+    // create delete span
+    const span = document.createElement('span');
+    span.className = 'del';
+    span.appendChild(document.createTextNode('Delete'));
+    // append button to div
+    div.appendChild(span);
+
+    desc.addEventListener('change', (e) => {
+      task.title = e.target.value;
+      // eslint-disable-next-line no-use-before-define
```
Comment (arslanbisharat) [inline src/index.js]: - [ ] Please don't disable linter rules. These are here to write clean and standard code.

File: src/index.js [hunk 9]
```diff
@@ -1,35 +1,163 @@
...
+  const data = window.localStorage.getItem('tasks');
+  if (data) {
+    const tasks = JSON.parse(data); // convert into an object
+    addElementsToPageFrom(tasks);
+  }
+}
+
+function deleteTaskWith(taskId) {
+  // filter //taskId = e.target.parentelemnt.getAttribute(data-id);
+  // eslint-disable-next-line eqeqeq
```
Comment (arslanbisharat) [inline src/index.js:124]: - [ ] Please don't disable linter rules. These are here to write clean and standard code.

File: src/index.js [hunk 10]
```diff
@@ -1,35 +1,163 @@
...
+  arrayOfTasks = arrayOfTasks.filter((task) => task.id != taskId);
+  addDataToLocal(arrayOfTasks);
+}
+
+function toggleStatusTask(taskId) {
+  for (let i = 0; i < arrayOfTasks.length; i += 1) {
+    // eslint-disable-next-line eqeqeq
```
Comment (arslanbisharat) [inline src/index.js:131]: - [ ] Please don't disable linter rules. These are here to write clean and standard code.

File: src/index.js [hunk 11]
```diff
@@ -1,35 +1,163 @@
...
+    if (arrayOfTasks[i].id == taskId) {
+      // eslint-disable-next-line no-unused-expressions
```
Comment (arslanbisharat) [inline src/index.js:133]: - [x] Please don't disable linter rules. These are here to write clean and standard code.

File: src/index.js [hunk 12]
```diff
@@ -1,35 +1,163 @@
...
+      arrayOfTasks[i].completed == false
+        ? (arrayOfTasks[i].completed = true)
+        : (arrayOfTasks[i].completed = false);
+    }
+  }
+  addDataToLocal(arrayOfTasks);
+}
+
+deletebtn.addEventListener('click', () => {
+  // tasksDiv.innerHTML = " ";
```
Comment (arslanbisharat) [inline src/index.js]: - [x] Remove commented code,

File: src/index.js [hunk 13]
```diff
@@ -1,35 +1,163 @@
...
+  // eslint-disable-next-line no-use-before-define
```
Comment (arslanbisharat) [inline src/index.js]: - [ ] Please don't disable linter rules. These are here to write clean and standard code.

File: src/index.js [hunk 14]
```diff
@@ -1,35 +1,163 @@
...
+  deleteTasks();
+  //  tasksDiv.innerHTML = " ";
```
Comment (arslanbisharat) [inline src/index.js]: - [x] Remove commented code.

File: src/style.css [hunk 15]
```diff
@@ -7,24 +11,129 @@ body {
   display: flex;
   flex-direction: column;
   align-items: center;
-  border: 2px solid black;
+
+  /* border: 2px solid black; */
```
Comment (arslanbisharat) [inline src/style.css]: - [x] remove commented code

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by addod19 — APPROVED** [review]
### Hi @Reem-lab, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**